scripts/lib/google-sheets-exporter.ts

```diff
@@ -0,0 +1,219 @@
+  backgroundColorStyle: {
+    rgbColor: {
+      red: 208 / 255,
+      green: 224 / 255,
+      blue: 227 / 255,
+      alpha: 1,
+    },
+  },
+};
+
+function createCellValue(
+  colName: string,
+  colValue: object | string | number | boolean,
+): sheets_v4.Schema$ExtendedValue {
+  let valToWrite: sheets_v4.Schema$ExtendedValue = {};
+  if (Array.isArray(colValue)) {
+    valToWrite = { stringValue: colValue.join(', ') };
+  } else if (typeof colValue === 'string') {
+    valToWrite = { stringValue: colValue };
+  } else if (typeof colValue === 'boolean') {
+    valToWrite = { boolValue: colValue };
+  } else if (typeof colValue === 'number') {
+    valToWrite = { numberValue: +colValue };
+  } else if (typeof colValue === 'object') {
+    throw new Error(
+      `Trying to write object to spreadsheet. Objects should be flattened prior to write. Column name: ${colName}; value: ${util.inspect(
+        colValue,
+      )}`,
+    );
+  } else {
+    console.warn(
+      `Unexpected value type: ${colName}: ${util.inspect(
+        colValue,
+      )}. Writing as string.`,
+    );
+    valToWrite = { stringValue: colValue };
+  }
+  return valToWrite;
+}
+
+export function collectedIncentiveToGoogleSheet(
+  incentives: CollectedIncentive[],
+  fieldMappings: AliasMap,
+  useStandardHeader: boolean,
+) {
+  const spreadsheetData = collectedIncentivesToSpreadsheet(
+    incentives,
+    fieldMappings,
+  );
+  return spreadsheetToGoogleSheet(spreadsheetData, useStandardHeader);
+}
+
+export function spreadsheetToGoogleSheet(
+  spreadsheet: SpreadsheetData,
+  useStandardHeader: boolean,
+) {
+  const { records, headers } = spreadsheet;
+
+  let rowData: sheets_v4.Schema$RowData[] = [];
+  records.forEach(record => {
+    const rowValues: sheets_v4.Schema$CellData[] = [];
+    for (const col of headers) {
+      const cell: sheets_v4.Schema$CellData = {
+        userEnteredFormat: standardValueCellFormat,
+      };
+      if (col in record) {
+        cell.userEnteredValue = createCellValue(col, record[col]);
+      }
+      rowValues.push(cell);
+    }
+    rowData.push({ values: rowValues });
+  });
+  // Prepend the headers.
+  const headerRows: sheets_v4.Schema$RowData[] = [];
+  if (useStandardHeader) {
+    const optionalHeaderRow: sheets_v4.Schema$CellData[] = [];
+    for (let i = 0; i < headers.length; i++) {
+      if (i === REBATE_AMOUNT_FIELDS_START) {
+        optionalHeaderRow.push({
+          userEnteredValue: {
+            stringValue: 'Rebate Amount Fields (guidelines)',
+          },
+          userEnteredFormat: { ...headerBaseFormat, ...rebateFieldsColor },
+          textFormatRuns: [
+            {
+              startIndex: 22,
+              format: {
+                link: {
+                  uri: 'https://docs.google.com/document/d/1nM9320uOUYpfpD3eZ4DFo32Scd5kauLEL17j1NYB5Ww/edit#bookmark=id.s9a0j7v2m6vb',
+                },
+              },
+            },
+            {
+              startIndex: 32,
```

Comment:
These two `startIndex` values should use the constants defined up top

Reply:
These are actually unrelated – they are the start/end indexes of `guidelines` in `Rebate Amount Fields (guidelines)` to properly display the hyperlink (that one's easy enough since those fields are hard-coded anyway).

Reply:
I'll add a comment to make that clearer.

Reply:
Actually I decided to do some string parsing – less brittle than the hard-coded indices (though only slightly).